tests/session/session_error-002.phpt

```diff
@@ -1,24 +0,0 @@
---TEST--
-MongoDB\Driver\Session with wrong defaultTransactionOptions
```

Comment:
I explain why I deleted this in the commit message:

> The deleted error test was redundant. It was originally created in ca4c52e to work around changes in PHP 8, but 60febd0 subsequently reintroduced a portable version of the test case.

This test was already covered by `session_error-001.phpt`, which this PR renamed to `manager-startSession_error-001.phpt`.